frontend/src/api/Search.js

```diff
@@ -1,137 +1,151 @@
-      var formattedArray = [];
-      
-      for (const word of wordsArray) {
-        // TODO: add more info to the word object, and checking for false domain matches
-        formattedArray.push({
-          word: word.lemma_wordform.text, 
-          slug: word.lemma_wordform.slug,
-          definitions: word.lemma_wordform.definitions,
-          });
-      }
-      
-      return formattedArray;
-    }
-  ).catch((error) => {
-    console.log(error);
-    return null;
-  });
+// Function to get hypernyms for a given index
+export function getHypernyms(index) {
+  if (!index || !rapidwordsData[index]) return [];
+
+  const hypernyms = rapidwordsData[index].hypernyms || [];
+  const formattedHypernyms = hypernyms.map(hypernym => [
+    hypernym,
+    rapidwordsData[hypernym]?.domain || "Unknown"
+  ]);
+
+  // Log hypernyms
+  logData("Hypernyms", formattedHypernyms);
+
+  return formattedHypernyms;
 }
 
-// a function that mocks the searchRWByDomain function
-// this is used for testing purposes
-// should return the list ['a', 'b', 'c'] as response
-export function searchRWByDomainMock(domain){
-  if (!domain) return;
-  
-  return new Promise((resolve, reject) => {
-    setTimeout(() => resolve(['a', 'b', 'c']), 1000);
-  });
+// Function to get hyponyms for a given index
+export function getHyponyms(index) {
+  if (!index || !rapidwordsData[index]) return [];
+
+  const hyponyms = rapidwordsData[index].hyponyms || [];
+  const formattedHyponyms = hyponyms.map(hyponym => [
+    hyponym,
+    rapidwordsData[hyponym]?.domain || "Unknown"
+  ]);
+
+  // Log hyponyms
+  logData("Hyponyms", formattedHyponyms);
+
+  return formattedHyponyms;
 }
 
-//{search('dog').then(res => {console.log(res)})}
-export function search(query){
-  try {
-      return axios.get('https://api.itwewina.altlab.dev/api/search/',
-        {params: {
-          name: query
-        }}
-        ).then(res => res.data)
-    }
+// Function to search RW by domain
+export function searchRWByDomain(index) {
+  if (!index) return;
 
-    catch (error) {
+  return axios
+    .get('https://itwewina.altlab.dev/api/rapidwords-index/', {
+      params: { rw_index: index }
+    })
+    .then((response) => {
+      // Get the results from the API response
+      const wordsArray = response.data.results;
+      
+      // Format the results with selected fields
+      const formattedArray = wordsArray.map((word) => ({
+        word: word.lemma_wordform.text,
+        slug: word.lemma_wordform.slug,
+        definitions: word.lemma_wordform.definitions,
+      }));
+
+      //logData("Search RW By Domain - Formatted Array", formattedArray);
+
+      // Return an object containing both the formatted results and the Cree words
+      return formattedArray;
+    })
+    .catch((error) => {
       console.log(error);
+      return null;
+    });
+}
+
+// Function to search for the index of a domain name
+export function searchDomainIndex(domainName) {
```

Comment:
This O(n) function is not the best approach, you can use a dictionary/hashmap data structure instead.